packages/core/src/utils.ts

```diff
@@ -259,3 +259,64 @@ export function errorAsyncIterator<TInput>(
 
   return stream;
 }
+
+export interface BasicCache<V> {
+  get(key: string): V | undefined;
+  set(key: string, value: V): void;
+  delete(key: string): void;
+  clear?: () => void;
+}
+
+export function getInMemoryLRUCache<V>({
+  max = 1000,
+  onDelete = () => {},
+}: {
+  max?: number;
+  onDelete?: (key: string, value: V) => void;
+} = {}): BasicCache<V> & { clear(): void } {
+  let storage: Record<string, V> = {};
+  let keys: string[] = [];
+  return {
+    get(key: string): V | undefined {
+      return storage[key];
+    },
+    set(key: string, value: V) {
+      queueMicrotask(() => {
+        if (storage[key] != null) {
+          if (keys.indexOf(key)) {
+            keys.splice(keys.indexOf(key), 1);
```

Comment:
maybe store the value of `keys.indexOf(key)` instead of running it again? :D

Reply:
Yes 😅  you're right. Done!